sharding/nodesCoordinator/indexHashedNodesCoordinatorLite_test.go

```diff
@@ -13,42 +14,69 @@ import (
 func createDummyValidatorsInfo() []*state.ShardValidatorInfo {
 	return []*state.ShardValidatorInfo{
 		{
-			PublicKey:  []uint8{1, 2, 3, 4},
+			PublicKey:  []byte("pubKey1"),
 			ShardId:    0,
 			List:       "eligible",
 			Index:      0,
 			TempRating: 10},
 		{
-			PublicKey:  []uint8{2, 3, 4, 5},
+			PublicKey:  []byte("pubKey2"),
 			ShardId:    0,
 			List:       "eligible",
 			Index:      1,
-			TempRating: 10},
+			TempRating: 11},
 		{
-			PublicKey:  []uint8{3, 4, 5, 1},
+			PublicKey:  []byte("pubKey3"),
 			ShardId:    0,
 			List:       "eligible",
 			Index:      2,
-			TempRating: 10},
+			TempRating: 12},
 		{
-			PublicKey:  []uint8{4, 5, 1, 2},
+			PublicKey:  []byte("pubKey4"),
 			ShardId:    core.MetachainShardId,
 			List:       "eligible",
 			Index:      0,
-			TempRating: 10},
+			TempRating: 13},
 		{
-			PublicKey:  []uint8{5, 1, 2, 3},
+			PublicKey:  []byte("pubKey5"),
 			ShardId:    core.MetachainShardId,
 			List:       "eligible",
 			Index:      1,
-			TempRating: 10},
+			TempRating: 14},
 		{
-			PublicKey:  []uint8{1, 2, 3, 5},
+			PublicKey:  []byte("pubKey6"),
 			ShardId:    core.MetachainShardId,
 			List:       "eligible",
 			Index:      2,
-			TempRating: 10},
+			TempRating: 15},
+	}
+}
+
+func verifyEligibleValidators(
+	t *testing.T,
+	validatorsInfo []*state.ShardValidatorInfo,
+	nodesConfig *epochNodesConfig,
+) {
+	if nodesConfig == nil {
```

Comment:
Please consider the following version.
Also, since from `epochNodesConfig` input param we only use `nodesConfig.eligibleMap`, perhaps we could refactor the func to only use `eligibleMap`

```go
	require.NotNil(t, nodesConfig)

	numEligibleValidators := 0
	for _, validatorsInShard := range nodesConfig.eligibleMap {

		for _, val := range validatorsInShard {
			found := false
			for _, validatorInfo := range validatorsInfo {
				samePubKey := bytes.Equal(val.PubKey(), validatorInfo.GetPublicKey())
				sameIndex := val.Index() == validatorInfo.GetIndex()
				if samePubKey && sameIndex {
					found = true
				}
			}
			assert.True(t, found)
			numEligibleValidators += 1
		}
	}

	assert.Equal(t, len(validatorsInfo), numEligibleValidators)
```